pkg/bgpv1/agent/controller.go

```diff
@@ -159,13 +160,24 @@ func configureForClusterPoolIPAM(factory externalversions.SharedInformerFactory,
 		UpdateFunc: func(_ interface{}, _ interface{}) { sig.Event(struct{}{}) },
 		DeleteFunc: sig.Event,
 	})
-	go nodeInformer.Run(stop)
 	return &ciliumNodeSpecer{
-		Name:       name,
-		NodeLister: nodeLister,
+		Name:         name,
+		NodeLister:   nodeLister,
+		nodeInformer: nodeInformer,
 	}, nil
 }
 
+// ControllerParams contains all parameters needed to construct a Controller
+type ControllerParams struct {
+	cell.In
+
+	Lifecycle    hive.Lifecycle
+	Clientset    client.Clientset
+	RouteMgr     BGPRouterManager
+	DaemonConfig *option.DaemonConfig
+	Opts         []ControllerOpt `group:"bgp-controller-options"`
```

Comment:
Ah, ControllerOpt isn't used at all. I don't think there's a reason to "future proof" this as only users of this are within the same repository, so I would suggest dropping it altogether.

Reply:
Yes, the `ControllerOpt`s were already in place, just refactored it to the new pattern. It might be for enterprise, @ldelossa what is your take on this?

Reply:
Not used. The API is just stamped out for future options. 

It seems plausible that as features are added to the controller, there will be options you'll want to apply to it. 

But I guess, it can just be added back at that time. 

It's not used in enterprise neither.